src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2369,6 +2369,15 @@ void Schedule::handleGRUPTREE( const DeckKeyword& keyword, size_t currentStep, c
         return groups.count(groupName) > 0;
     }
 
+    bool Schedule::hasGroup(const std::string& groupName, std::size_t timeStep) const {
+        if (timeStep >= this->size())
+            return false;
+
+        auto grpMap = this->groups.find(groupName);
+
+        return (grpMap != this->groups.end())
+            && (grpMap->second.at(timeStep) != nullptr);
```

Comment:
I was surprised to learn that `std::shared_ptr` compared the content with `nullptr` - that was in my opinion bad taste from some committee - if you agree with me you could do:
```
return ... && (grpMap->second.at(timeStep));
```

Reply:
I suppose it's a special concession to making `std::shared_ptr<T>` behave as much as a `T*` as possible. For that matter `std::unique_ptr<T>` has the same behaviour.  I too was surprised when I learned about it.  Since there is indeed risk that the reader will be confused, I'll remove the explicit `nullptr` here.